CHANGELOG.md

```diff
@@ -35,6 +35,7 @@
   * `cortex_ingester_tsdb_block_postings_for_matchers_cache_misses_total`
   * `cortex_ingester_tsdb_block_postings_for_matchers_cache_requests_total`
   * `cortex_ingester_tsdb_block_postings_for_matchers_cache_skips_total`
+* [ENHANCEMENT] Compactor: Shuffle order of users in `UsersScanner` before submitting to `BlocksCleaner`. #10513
```

Comment:
Can this briefly say "why" in addition to "how"? So that unfamiliar readers understand why they might upgrade, and so on. :)